tizen/src/tasks.h

```diff
@@ -0,0 +1,111 @@
+#ifndef FLUTTER_PLUGIN_WORKMANAGER_TASKS_H_
+#define FLUTTER_PLUGIN_WORKMANAGER_TASKS_H_
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include "options.h"
+
+class WorkManagerCall {
+   public:
+    WorkManagerCall(){};
+};
+
+class InitializeTask : public WorkManagerCall {
+   public:
+    InitializeTask(int32_t callBackDispathcerHandlerKey, bool isInDebugMode)
+        : callBackDispathcerHandlerKey(callBackDispathcerHandlerKey),
+          isInDebugMode(isInDebugMode){};
+
+    int32_t callBackDispathcerHandlerKey = 0;
+    bool isInDebugMode = false;
+};
+
+class RegisterTask : public WorkManagerCall {
+   public:
+    RegisterTask(bool isInDebugMode, std::string uniquename,
+                 std::string taskname, std::optional<std::string> tag,
+                 int32_t initialDelaySeconds,
+                 std::optional<Constraints> constraintsConfig,
+                 std::optional<std::string> payload)
+        : isInDebugMode(isInDebugMode),
+          uniquename(uniquename),
+          taskname(taskname),
+          tag(tag),
+          initialDelaySeconds(initialDelaySeconds),
+          constraintsConfig(constraintsConfig),
+          payload(payload){};
+
+    bool isInDebugMode;
+    std::string uniquename;
+    std::string taskname;
+    std::optional<std::string> tag;
+    int32_t initialDelaySeconds;
+    std::optional<Constraints> constraintsConfig;
+    std::optional<std::string> payload;
+};
+
+class OneoffTask : public RegisterTask {
+   public:
+    ExistingWorkPolicy existingWorkPolicy;
+    std::optional<BackoffPolicyTaskConfig> backoffPolicyConfig;
+    std::optional<OutOfQuotaPolicy> outOfQuotaPolicy;
```

Comment:
Is there a reason why the member variables are public?
The member variable of other classes is also public, is it intended?

Reply:
They can be accessed by other classes so I let them public.

Reply:
You should not be able to access member variables of other classes directly from other classes. You must access the variable using the getter and setter for it.

Reply:
I changed them into struct.